src/SLCore/Analysis/SLCoreRuleSettings.cs

```diff
@@ -0,0 +1,81 @@
+﻿/*
+ * SonarLint for Visual Studio
+ * Copyright (C) 2016-2024 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System.ComponentModel.Composition;
+using SonarLint.VisualStudio.Core;
+using SonarLint.VisualStudio.Core.UserRuleSettings;
+using SonarLint.VisualStudio.SLCore.Core;
+using SonarLint.VisualStudio.SLCore.Service.Rules;
+using SonarLint.VisualStudio.SLCore.Service.Rules.Models;
+
+namespace SonarLint.VisualStudio.SLCore.Analysis;
+
+public interface ISLCoreRuleSettings
+{
+    Dictionary<string, StandaloneRuleConfigDto> RulesSettings { get; }
+    void UpdateStandaloneRulesConfiguration();
+}
+
+[Export(typeof(ISLCoreRuleSettings))]
+[PartCreationPolicy(CreationPolicy.Shared)]
+internal sealed class SlCoreRuleSettings : ISLCoreRuleSettings
+{
+    private readonly ILogger logger;
+    private readonly IUserSettingsProvider userSettingsProvider;
+    private readonly ISLCoreServiceProvider slCoreServiceProvider;
+
+    [ImportingConstructor]
+    public SlCoreRuleSettings(ILogger logger, IUserSettingsProvider userSettingsProvider, ISLCoreServiceProvider slCoreServiceProvider)
+    {
+        this.logger = logger;
+        this.userSettingsProvider = userSettingsProvider;
+        this.slCoreServiceProvider = slCoreServiceProvider;
+    }
+
+    public Dictionary<string, StandaloneRuleConfigDto> RulesSettings => MapRuleSettingsToSlCoreSettings(userSettingsProvider.UserSettings.RulesSettings);
```

Comment:
public -> private -> public -> private is hard to read

Reply:
Can you be more explicit what are you referring to, please?

Reply:
The method ordering. You have a public method, followed by a private method, followed by a public method and then a private method again

Reply:
We should talk about in the weekly

Reply:
Will be done as part of the architecture refactoring, in https://sonarsource.atlassian.net/browse/SLVS-1301